kernel/src/platform/watchdog.rs

```diff
@@ -0,0 +1,27 @@
+//! Interface for configuring a watchdog
+
+/// A trait for implementing a watchdog in the kernel.
+/// This trait is called from the `kernel_loop()` code to setup
+/// and maintain the watchdog timer.
+/// It is up to the specific `Chip` how it will handle watchdog interrupts.
+pub trait WatchDog {
+    /// This function must enable the watchdog timer and configure it to
+    /// trigger regulary. The period of the timer is left to the implementation
+    /// to decide. The implementation must ensure that it doesn't trigger too
+    /// early (when we haven't hung for example) or too late as to not catch
+    /// faults.
+    fn setup(&self) {}
```

Comment:
Is the watchdog running once `setup` returns, or does it start running after the first `tickle`?

Reply:
I have changed this to specify that the timer must be running